src/btreemap/node/io.rs

```diff
@@ -246,32 +255,29 @@ impl<'a, M: Memory> NodeWriter<'a, M> {
             Bytes::from(self.page_size.get()),
         );
 
-        let mut bytes_written = 0;
+        let mut position = 0;
         for RealSegment {
             page_idx,
             offset,
             length,
         } in iter
         {
-            let offset = if page_idx == 0 {
-                // Write to the initial page.
-                (self.address + offset).get()
-            } else {
-                // Write to an overflow page, allocating it if it doesn't already exist.
-                if self.overflows.len() < page_idx {
-                    self.allocate_new_page();
-                }
-
-                (self.overflows[page_idx - 1] + offset).get()
+            let len = length.get() as usize;
+            match page_idx {
```

Comment:
Why don't we use this match to compute the offset and then perform the write separately afterwards (to avoid code duplication)?

Reply:
I agree that this would be easier to read, and I tried it, but it turned out to be a bit slower, somewhere around +0.16% or so.